jps-plugin/src/org/jetbrains/jps/incremental/scala/local/CompilerFactoryImpl.scala

```diff
@@ -97,11 +97,24 @@ object CompilerFactoryImpl {
                                client: Option[Client]): File = {
 
     val scalaVersion = scalaInstance.actualVersion
+    def getSourceJars = if (isBefore_2_11(scalaVersion)) sourceJars._2_10 else sourceJars._2_11
+
     val sourceJar =
-      if (isBefore_2_11(scalaVersion)) sourceJars._2_10
-      else sourceJars._2_11
+      if (scalaVersion.contains("hydra")) {
+        val hydraBridge = scalaInstance.otherJars().find(_.getName.contains(HydraData.HydraBridgeName))
+        if(hydraBridge.isEmpty)
+          Log.warn(s"Hydra Bridge was not found for $scalaVersion " +
+            s"in ${scalaInstance.otherJars().map(_.getName).mkString(",")}. The vanilla bridge will be used instead.")
+        hydraBridge.getOrElse(getSourceJars)
+      }
+      else getSourceJars
+
+    def getHydraVersion = HydraData.getHydraVersionFromBridge(sourceJar) match {
+      case Some(ver) => s"-$ver"
+      case _ => ""
+    }
 
-    val interfaceId = "compiler-interface-" + scalaVersion + "-" + javaClassVersion
+    val interfaceId = "compiler-interface-" + scalaVersion + getHydraVersion + "-" + javaClassVersion
```

Comment:
Why is there a call to `getHydraVersion` here? We don't publish an artifact named "compiler-interface" so what's the interest of changing this?

Reply:
A jar is created using the interfaceId as a name. It is not enough (in our case) to provide only the Scala version as a discriminant, we also need the Hydra version. For example, we are using Scala version 2.11.8 and Hydra 0.9.4 and now the jar file is created, but if we change Hydra to 0.9.5 the jar "already exists" because interfaceId is the same in both cases.

Reply:
Can you tell me what's the difference between the compiler-interface for Hydra 0.9.4 versus 0.9.5, for a specific Scala version? I'm asking because in sbt we don't need a different compiler-interface, and I would expect the same inside IntelliJ.

Reply:
I don't really know what is different, but I see this call to compileSources and the parameter sourceJar. The sourceJar is HydraBridge in our case, so I see that there are different bridges for 0.9.4 and for 0.9.5 ( or maybe they are the same but the name is different)  so because of this I assumed that I must compile the sources again if the Hydra Version is different.
Is this ok or should I change it to how it was before? 

`AnalyzingCompiler.compileSources(sourceJar :: Nil, targetJar, interfaceJars, interfaceId, raw, NullLogger)`

Reply:
Yes, please, change it and check if there is any problem. I don't expect any, but who knows.

Reply:
It worked. I pushed the changes.